### PR TITLE
[DEVOPS-61] appveyor cache

### PR DESCRIPTION
**Pull request by domenkozar**
Depends on https://github.com/input-output-hk/iohk-ops/pull/221

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -8,49 +8,32 @@ environment:
     # Avoid long paths on Windows
     STACK_ROOT: "c:\\s"
     STACK_WORK: ".w"
-    CACHE_DIR: "c:\\c"
     WORK_DIR: "c:\\w"
+    CACHE_S3_VERSION: v0.1.0
+    AWS_REGION: us-west-1
+    BUCKET: appveyor-ci-cache
```
Comment (lehins) [inline appveyor.yml]: `cache-s3` expects this variable's name to be `S3_BUCKET` instead of `BUCKET`

File: appveyor.yml [hunk 2]
```diff
@@ -8,49 +8,32 @@ environment:
     # Avoid long paths on Windows
     STACK_ROOT: "c:\\s"
     STACK_WORK: ".w"
-    CACHE_DIR: "c:\\c"
     WORK_DIR: "c:\\w"
+    CACHE_S3_VERSION: v0.1.0
```
Comment (lehins) [inline appveyor.yml]: There is newer release of `cache-s3`: `v0.1.1`, that has an additional feature `--public`, which has no effect cardano-sl, so leaving it at `v0.1.0` is fine as well.

**Comment by lehins** [comment]
@domenkozar Besides the variable name this looks great.

**Comment by lehins** [comment]
Changes in 416d95f look good!. :+1:

File: appveyor.yml [hunk 3]
```diff
@@ -8,49 +8,32 @@ environment:
     # Avoid long paths on Windows
     STACK_ROOT: "c:\\s"
     STACK_WORK: ".w"
-    CACHE_DIR: "c:\\c"
     WORK_DIR: "c:\\w"
+    CACHE_S3_VERSION: v0.1.1
+    AWS_REGION: us-west-1
```
Comment (deepfire) [inline appveyor.yml]: @domenkozar, do we really want to use `us-west-1`?  I think something EU-based like `eu-central-1`, or alternatively `eu-west-1` (for older auth protocol version support, if needed) would make more sense.

File: appveyor.yml [hunk 4]
```diff
@@ -8,49 +8,32 @@ environment:
...
+    S3_BUCKET: appveyor-ci-cache              
+    AWS_ACCESS_KEY_ID: 
+      secure: 0OtMCUmJ4V7fNRPvKjzYxSO19I200rPG8WRW/qn1QPk=
```
Comment (deepfire) [inline appveyor.yml]: @domenkozar, do we have it documented somewhere, what user we're using for accessing the cache bucket?

File: appveyor.yml (hunk 3 above)
Comment (lehins) [reply appveyor.yml]: @deepfire It's better to place the bucket in the region closest to AppVeyor servers, so the cache can be uploaded/downloaded quicker. I don't know if @domenkozar decided on `us-west-1`, experimentally, but considering that AppVeyor is registered in Canada, I'd expect placing S3 bucket in Europe would be slower.

File: appveyor.yml (hunk 4 above)
Comment (lehins) [reply appveyor.yml]: @deepfire Don't know about the documentation part, but the user is created by terraform here: https://github.com/input-output-hk/iohk-ops/pull/221/files#diff-fefc5952bb2ab240eaa7c2245608fc79R7 and it's default name will be `appveyor-ci-cache-user`

File: appveyor.yml (hunk 3 above)
Comment (deepfire) [reply appveyor.yml]: Understood, makes sense, thanks!

File: appveyor.yml (hunk 4 above)
Comment (deepfire) [reply appveyor.yml]: Thanks again!

**Comment by deepfire** [comment]
@domenkozar, I've hit the "rebuild PR" button on both the PR and branch builds, and yet both came at 40+ minutes.

**Comment by domenkozar** [comment]
@deepfire this won't take effect until we merge it. If you follow the logic through, it will never save state for a PR

**Comment by deepfire** [comment]
Oh, so it's not meant to reduce latency for branch merging -- only latency to first results, I see.

**Comment by lehins** [comment]
@deepfire it will reduce the latency for merging as well, except for PRs it will use cache from master, which is yet not available.

**Comment by domenkozar** [comment]
I've changed this logic to also support `develop` and `release` branches.

**Pull request merged**